core/camel-support/src/main/java/org/apache/camel/converter/stream/StreamCacheConverter.java

```diff
@@ -50,9 +50,11 @@ public static StreamCache convertToStreamCache(ByteArrayInputStream stream, Exch
     public static StreamCache convertToStreamCache(InputStream stream, Exchange exchange) throws IOException {
         // transfer the input stream to a cached output stream, and then creates a new stream cache view
         // of the data, which ensures the input stream is cached and re-readable.
-        CachedOutputStream cos = new CachedOutputStream(exchange);
-        IOHelper.copyAndCloseInput(stream, cos);
-        return cos.newStreamCache();
+        // Use try-with-resources to ensure the CachedOutputStream is closed automatically
+        try (CachedOutputStream cos = new CachedOutputStream(exchange)) {
```

Comment:
This is wrong as COS should not be closed - there is some special code for stream cahcing when its based on file disk and whatnot.